src/CommonITILObject.php

```diff
@@ -7770,6 +7770,73 @@ protected function updateActors()
        // reload actors
         $this->loadActors();
 
+        if (isset($this->input['_additional_observers'])) {
```

Comment:
Maybe code could be mutualized like this.
```php
$additionnal_actors_fields = [
    User::class => [
        'observer' => '_additional_observers',
        'requester' => '_additional_requesters',
        'assign' => '_additional_assigns',
    ],
    Group::class => [
        'observer' => '_additional_groups_observers',
        'requester' => '_additional_groups_requesters',
        'assign' => '_additional_groups_assigns',
    ],
];
foreach ($additionnal_actors_fields as $actor_class => $fields) {
    foreach ($fields as $actor_type => $field) {
        if (isset($this->input[$field])) {
            foreach ($this->input[$field] as $additional_actor) {
                $this->input['_actors'][$actor_type][] = [
                    'itemtype' => $actor_class::getType(),
                    'items_id' => $additional_actor[$actor_class::getForeignKeyField()],
                    'use_notification' => $additional_actor['use_notification'],
                    'alternative_email' => ''
                ];
            }
        }
    }
}
```

Reply:
refactor done